state/stateChanges/dataAnalysisCollector.go

```diff
@@ -163,22 +165,49 @@ func (scc *dataAnalysisCollector) Reset() {
 	scc.cachedTxs = make(map[string]*transaction.Transaction)
 }
 
-// Publish will export state changes
-func (scc *dataAnalysisCollector) Publish() error {
+// Publish will retrieve the state changes linked with the tx hash.
+func (scc *dataAnalysisCollector) Publish() (map[string]*data.StateChanges, error) {
```

Comment:
is this correct? this is the same as for normal collector.
my suggestion was to store into levelDB on this Publish, and avoid having `Store` function, since this component is used only for data analysis, not for the production flow, at least currently

Reply:
this will also save normal state changes to outport driver when running in data analysis mode, which i don't think we want

Reply:
yes, but in the current implementation if you don't have the outport driver configured you will never `publish`, hence the `store` function. If you have data analysis enabled and the host driver off, you will still have them persisted in leveldb.